src/emqttd_access_control.erl

```diff
@@ -123,89 +102,71 @@ register_mod(Type, Mod, Opts) when Type =:= auth; Type =:= acl->
 register_mod(Type, Mod, Opts, Seq) when Type =:= auth; Type =:= acl->
     gen_server:call(?SERVER, {register_mod, Type, Mod, Opts, Seq}).
 
-%%------------------------------------------------------------------------------
 %% @doc Unregister authentication or ACL module
-%% @end
-%%------------------------------------------------------------------------------
 -spec unregister_mod(Type :: auth | acl, Mod :: atom()) -> ok | {error, any()}.
 unregister_mod(Type, Mod) when Type =:= auth; Type =:= acl ->
     gen_server:call(?SERVER, {unregister_mod, Type, Mod}).
 
-%%------------------------------------------------------------------------------
-%% @doc Lookup authentication or ACL modules
-%% @end
-%%------------------------------------------------------------------------------
+%% @doc Lookup authentication or ACL modules.
 -spec lookup_mods(auth | acl) -> list().
 lookup_mods(Type) ->
     case ets:lookup(?ACCESS_CONTROL_TAB, tab_key(Type)) of
-        [] -> [];
+        []          -> [];
         [{_, Mods}] -> Mods
     end.
 
 tab_key(auth) -> auth_modules;
 tab_key(acl)  -> acl_modules.
 
-%%------------------------------------------------------------------------------
-%% @doc Stop access control server
-%% @end
-%%------------------------------------------------------------------------------
-stop() ->
-    gen_server:call(?MODULE, stop).
+%% @doc Stop access control server.
+stop() -> gen_server:call(?MODULE, stop).
 
-%%%=============================================================================
-%%% gen_server callbacks
-%%%=============================================================================
+%%--------------------------------------------------------------------
+%% gen_server callbacks
+%%--------------------------------------------------------------------
 
 init([Opts]) ->
     ets:new(?ACCESS_CONTROL_TAB, [set, named_table, protected, {read_concurrency, true}]),
-
     ets:insert(?ACCESS_CONTROL_TAB, {auth_modules, init_mods(auth, proplists:get_value(auth, Opts))}),
     ets:insert(?ACCESS_CONTROL_TAB, {acl_modules, init_mods(acl, proplists:get_value(acl, Opts))}),
-    {ok, state}.
+    {ok, #state{}}.
 
 init_mods(auth, AuthMods) ->
-    [init_mod(fun authmod/1, Name, Opts) || {Name, Opts} <- AuthMods];
+    [init_mod(authmod(Name), Opts) || {Name, Opts} <- AuthMods];
 
 init_mods(acl, AclMods) ->
-    [init_mod(fun aclmod/1, Name, Opts) || {Name, Opts} <- AclMods].
+    [init_mod(aclmod(Name), Opts) || {Name, Opts} <- AclMods].
 
-init_mod(Fun, Name, Opts) ->
-    Module = Fun(Name),
-    {ok, State} = Module:init(Opts),
-    {Module, State, 0}.
+init_mod(Mod, Opts) ->
+    {ok, State} = Mod:init(Opts), {Mod, State, 0}.
```

Comment:
Remove the dynamic function call on line 141. Only modules that define callbacks should make dynamic calls.